### PR TITLE
Quick fix for interface and transaction versioning issues

### PR DESCRIPTION
**Pull request by carlpulley-da**
CHANGELOG_BEGIN

* Ensure SBUBeginExercise has a version that is max(template version, interface version)

CHANGELOG_END

- [x] implement a quick fix for issue
- [x] merged testing from PR #14765

~~Future PRs will add in testing (see https://github.com/digital-asset/daml/pull/14765) for this issue.~~

### CODE REVIEW

**Review by remyhaemmerle-da — COMMENTED** [review]
Thank.

I will change the description though. It is not really a bug fix, but rather a non implemented feature.

**Review by cocreature — COMMENTED** [review]
lgtm, can we get a test for this? I’d like to see:

1. template version newer than interface
2. interface newer than template
3. and i guess for completeness: both have the same version

**Comment by carlpulley-da** [comment]
> lgtm, can we get a test for this? I’d like to see:
> 
> 1. template version newer than interface
> 2. interface newer than template
> 3. and i guess for completeness: both have the same version

NP. I'll factor testing out into a separate PR.

**Comment by carlpulley-da** [comment]
Testing for this PR is under https://github.com/digital-asset/daml/pull/14765.

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/TransactionVersionTest.scala [hunk 1]
```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (carlpulley-da) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/TransactionVersionTest.scala:1]: Changes from PR #14765

**Review by cocreature — APPROVED** [review]
lgtm thx!

**Pull request merged**